oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtTimestampValidator.java

```diff
@@ -65,20 +67,33 @@ public JwtTimestampValidator() {
 
 	public JwtTimestampValidator(Duration clockSkew) {
 		Assert.notNull(clockSkew, "clockSkew cannot be null");
+		this.allowEmpty = true;
 		this.clockSkew = clockSkew;
 	}
 
+	/**
+	 * Whether to allow the {@code exp} or {@code nbf} header to be empty. The default value is
+	 * {@code true}
+	 */
+	public void setAllowEmpty(boolean allowEmpty) {
```

Comment:
I think this needs to be more flexible, given that there are two different claims. It seems reasonable to say that `exp` is required while `nbf` is not.

Will you please change this to be `setAllowEmptyExpiryClaim` and `setAllowEmptyNotBeforeClaim`? These would each have their own toggle and be `true` by default.